docs/content/ingestion/batch-ingestion.md

```diff
@@ -166,6 +166,60 @@ s3n://billy-bucket/the/data/is/here/y=2012/m=06/d=01/H=01
 ...
 s3n://billy-bucket/the/data/is/here/y=2012/m=06/d=01/H=23
 ```
+##### `datasource`
+
+It is a type of inputSpec that reads data already stored inside druid. It is useful for doing "re-indexing". A usecase would be that you ingested some data in some interval and at a later time you wanted to change granularity of rows or remove some columns from the data stored in druid.
+
+|Field|Type|Description|Required|
+|-----|----|-----------|--------|
+|ingestionSpec|Json Object|Specification of druid segments to be loaded. See below.|yes|
+|maxSplitSize|Number|Enables combining multiple segments into single Hadoop InputSplit according to size of segments. Default is none. |no|
+
+Here is what goes inside "ingestionSpec"
+|Field|Type|Description|Required|
+|datasource|String|Druid datasource name from which you are loading the data.|yes|
+|interval|String|A string representing representing ISO-8601 Intervals.|yes|
```

Comment:
you mean interval in "granularitySpec".
In general, I felt like it wasn't a good idea to mix the two. Also interface GranularitySpec does not really have a method called getInterval()
also this pathSpec has support for one interval only.
I would keep it like this for now and don't want to put a note in the doc to not confuse user with additional technical details. We can change it later if needed.